Week 12/compound/src/FlashLoan.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.20;
+
+import {console2} from "../lib/forge-std/src/Test.sol";
+import {IERC20} from "../lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {ERC20} from "../lib/openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
+import {CToken} from "../lib/compound-protocol/contracts/CToken.sol";
+import {CErc20Delegator} from "../lib/compound-protocol/contracts/CErc20Delegator.sol";
+import {
+    IFlashLoanSimpleReceiver,
+    IPoolAddressesProvider,
+    IPool
+} from "aave-v3-core/contracts/flashloan/interfaces/IFlashLoanSimpleReceiver.sol";
+
+interface ISwapRouter {
+    struct ExactInputSingleParams {
+        address tokenIn;
+        address tokenOut;
+        uint24 fee;
+        address recipient;
+        uint256 deadline;
+        uint256 amountIn;
+        uint256 amountOutMinimum;
+        uint160 sqrtPriceLimitX96;
+    }
+
+    function exactInputSingle(ISwapRouter.ExactInputSingleParams memory params) external returns (uint256 amountOut);
+}
+
+// TODO: Inherit IFlashLoanSimpleReceiver
+contract AaveFlashLoan is IFlashLoanSimpleReceiver {
+    address constant UNI = 0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984;
+    address constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address constant POOL_ADDRESSES_PROVIDER = 0x2f39d218133AFaB8F2B819B1066c7E434Ad94E9e;
+    ISwapRouter swapRouter = ISwapRouter(0xE592427A0AEce92De3Edee1F18E0157C05861564);
+
+    function execute(bytes calldata data) external {
+        (,,,, uint256 liquidateAmount) = abi.decode(data, (CErc20Delegator, CErc20Delegator, address, address, uint256));
+        // console2.log("5. Get 1250 USDC from AAVE to AaveFlashLoan contract ");
+        IPool(address(POOL())).flashLoanSimple(address(this), USDC, liquidateAmount, data, 0);
+    }
+
+    function executeOperation(address asset, uint256 amount, uint256 premium, address initiator, bytes calldata params)
```

Comment:
Almost perfect, 2 checks are missing here:
1. msg.sender needs to be the aave pool
2. initiator needs to be you address